classes/npm.bbclass

```diff
@@ -208,50 +209,45 @@ python clean_npm() {
 do_cleanall[postfuncs] += "clean_npm"
 
 do_install() {
-    dpkg_do_mounts
-
-    # changing the home directory to the working directory, the .npmrc will
-    # be created in this directory
-    export HOME=${PP}
-
-    # ensure empty cache
-    export npm_config_cache=${PP}/npm_cache
-    sudo rm -rf ${WORKDIR}/npm_cache
+    # create directories to be installed
+    if [ -n "${NPM_LOCAL_INSTALL_DIR}" ]; then
+        mkdir -p ${D}/${NPM_LOCAL_INSTALL_DIR}
+    else
+        mkdir -p ${D}/usr/lib
+    fi
+}
 
+do_prepare_build_append() {
     INSTALL_FLAGS="--offline --only=production --no-package-lock --verbose \
                    --arch=${NPM_ARCH} --target_arch=${NPM_ARCH}"
 
     if [ -n "${NPM_LOCAL_INSTALL_DIR}" ]; then
-        mkdir -p ${D}/${NPM_LOCAL_INSTALL_DIR}
         CHDIR=${PP}/image/${NPM_LOCAL_INSTALL_DIR}
     else
         CHDIR=/
-        mkdir -p ${D}/usr/lib
         INSTALL_FLAGS="$INSTALL_FLAGS --prefix ${PP}/image/usr -g"
     fi
 
     if [ -n "${NPM_REBUILD}" ]; then
         INSTALL_FLAGS="$INSTALL_FLAGS --build-from-source --no-save"
     fi
 
-    export CHDIR INSTALL_FLAGS
-    sudo -E chroot --userspec=$(id -u):$(id -g) ${BUILDCHROOT_DIR} sh -c ' \
-        cd $CHDIR
-        npm install $INSTALL_FLAGS ${NPM_INSTALL_FLAGS} /downloads/${@get_npm_bundled_tgz(d)}
-    '
+    cat <<EOF >> ${S}/debian/rules
```

Comment:
Why not simply provide that rules file (as a template) instead of generating it on the fly?

Reply:
This is a class, not some recipe include with own files folder.

Reply:
Ok, understood. Then it should be fine.